Add the waypoint_data table

pangolp opened the pull request:
It's likely there will be some errors, as this is my first time working with the editor. The idea was to add the waypoint_data table. If you see any errors, please let me know so I can try to correct them. If everything goes well, the plan is to continue with the other three tables mentioned in the issue.

I haven't been able to test it on Windows because I'm getting an error. Although I have what you mention installed because I compile the emulator. And without that, it wouldn't compile; perhaps it's missing something else.

<details>
<summary>Error running npm install</summary>

```
npm warn skipping integrity check for git dependency ssh://git@github.com/electron/node-gyp.git
npm warn deprecated @types/electron-settings@4.0.2: This is a stub types definition. electron-settings provides its own type definitions, so you do not need this installed.
npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm warn deprecated @npmcli/move-file@1.1.2: This functionality has been moved to @npmcli/fs
npm warn deprecated @npmcli/move-file@2.0.1: This functionality has been moved to @npmcli/fs
npm warn deprecated npmlog@6.0.2: This package is no longer supported.
npm warn deprecated @humanwhocodes/config-array@0.13.0: Use @eslint/config-array instead
npm warn deprecated rimraf@2.6.3: Rimraf versions prior to v4 are no longer supported
npm warn deprecated rimraf@2.7.1: Rimraf versions prior to v4 are no longer supported
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated har-validator@5.1.5: this library is no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@8.1.0: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@8.1.0: Glob versions prior to v9 are no longer supported
npm warn deprecated are-we-there-yet@3.0.1: This package is no longer supported.
npm warn deprecated @humanwhocodes/object-schema@2.0.3: Use @eslint/object-schema instead
npm warn deprecated squel@5.13.0: No longer maintained
npm warn deprecated gauge@4.0.4: This package is no longer supported.
npm warn deprecated boolean@3.2.0: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
npm warn deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm warn deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm warn deprecated eslint@8.57.1: This version is no longer supported. Please see https://eslint.org/version-support for other options.
npm warn cleanup Failed to remove some directories [
npm warn cleanup   [
npm warn cleanup     '\\\\?\\C:\\Users\\Walter\\Desktop\\keira3\\node_modules\\@nx\\webpack\\node_modules',
npm warn cleanup     [Error: EPERM: operation not permitted, rmdir 'C:\Users\Walter\Desktop\keira3\node_modules\@nx\webpack\node_modules'] {
npm warn cleanup       errno: -4048,
npm warn cleanup       code: 'EPERM',
npm warn cleanup       syscall: 'rmdir',
npm warn cleanup       path: 'C:\\Users\\Walter\\Desktop\\keira3\\node_modules\\@nx\\webpack\\node_modules'
npm warn cleanup     }
npm warn cleanup   ],
npm warn cleanup   [
npm warn cleanup     '\\\\?\\C:\\Users\\Walter\\Desktop\\keira3\\node_modules\\@angular-devkit',
npm warn cleanup     [Error: EPERM: operation not permitted, rmdir 'C:\Users\Walter\Desktop\keira3\node_modules\@angular-devkit\build-angular\node_modules\@angular\build\node_modules'] {
npm warn cleanup       errno: -4048,
npm warn cleanup       code: 'EPERM',
npm warn cleanup       syscall: 'rmdir',
npm warn cleanup       path: 'C:\\Users\\Walter\\Desktop\\keira3\\node_modules\\@angular-devkit\\build-angular\\node_modules\\@angular\\build\\node_modules'
npm warn cleanup     }
npm warn cleanup   ],
npm warn cleanup   [
npm warn cleanup     '\\\\?\\C:\\Users\\Walter\\Desktop\\keira3\\node_modules\\@module-federation\\utilities',
npm warn cleanup     [Error: EPERM: operation not permitted, rmdir 'C:\Users\Walter\Desktop\keira3\node_modules\@module-federation\utilities\node_modules'] {
npm warn cleanup       errno: -4048,
npm warn cleanup       code: 'EPERM',
npm warn cleanup       syscall: 'rmdir',
npm warn cleanup       path: 'C:\\Users\\Walter\\Desktop\\keira3\\node_modules\\@module-federation\\utilities\\node_modules'
npm warn cleanup     }
npm warn cleanup   ]
npm warn cleanup ]
npm error code 1
npm error path C:\Users\Walter\Desktop\keira3\node_modules\sqlite3
npm error command failed
npm error command C:\Windows\system32\cmd.exe /d /s /c prebuild-install -r napi || node-gyp rebuild
npm error prebuild-install warn This package does not support N-API version undefined
npm error (node:3408) [DEP0176] DeprecationWarning: fs.R_OK is deprecated, use fs.constants.R_OK instead
npm error (Use `node --trace-deprecation ...` to show where the warning was created)
npm error prebuild-install warn install No prebuilt binaries found (target=undefined runtime=napi arch=x64 libc= platform=win32)
npm error gyp info it worked if it ends with ok
npm error gyp info using node-gyp@8.4.1
npm error gyp info using node@24.1.0 | win32 | x64
npm error (node:19224) [DEP0060] DeprecationWarning: The `util._extend` API is deprecated. Please use Object.assign() instead.
npm error (Use `node --trace-deprecation ...` to show where the warning was created)
npm error (node:19224) [DEP0190] DeprecationWarning: Passing args to a child process with shell option true can lead to security vulnerabilities, as the arguments are not escaped, only concatenated.
npm error gyp info find Python using Python version 3.12.4 found at "D:\Users\Walter\AppData\Local\Programs\Python\Python312\python.exe"
npm error gyp ERR! find VS
npm error gyp ERR! find VS msvs_version not set from command line or npm config
npm error gyp ERR! find VS VCINSTALLDIR not set, not running in VS Command Prompt
npm error gyp ERR! find VS checking VS2022 (17.11.35219.272) found at:
npm error gyp ERR! find VS "D:\Program Files\Microsoft Visual Studio\2022\Community"
npm error gyp ERR! find VS - found "Visual Studio C++ core features"
npm error gyp ERR! find VS - found VC++ toolset: v143
npm error gyp ERR! find VS - missing any Windows SDK
npm error gyp ERR! find VS could not find a version of Visual Studio 2017 or newer to use
npm error gyp ERR! find VS looking for Visual Studio 2015
npm error gyp ERR! find VS - not found
npm error gyp ERR! find VS not looking for VS2013 as it is only supported up to Node.js 8
npm error gyp ERR! find VS
npm error gyp ERR! find VS **************************************************************
npm error gyp ERR! find VS You need to install the latest version of Visual Studio
npm error gyp ERR! find VS including the "Desktop development with C++" workload.
npm error gyp ERR! find VS For more information consult the documentation at:
npm error gyp ERR! find VS https://github.com/nodejs/node-gyp#on-windows
npm error gyp ERR! find VS **************************************************************
npm error gyp ERR! find VS
npm error gyp ERR! configure error
npm error gyp ERR! stack Error: Could not find any Visual Studio installation to use
npm error gyp ERR! stack     at VisualStudioFinder.fail (C:\Users\Walter\Desktop\keira3\node_modules\sqlite3\node_modules\node-gyp\lib\find-visualstudio.js:122:47)
npm error gyp ERR! stack     at C:\Users\Walter\Desktop\keira3\node_modules\sqlite3\node_modules\node-gyp\lib\find-visualstudio.js:75:16
npm error gyp ERR! stack     at VisualStudioFinder.findVisualStudio2013 (C:\Users\Walter\Desktop\keira3\node_modules\sqlite3\node_modules\node-gyp\lib\find-visualstudio.js:363:14)
npm error gyp ERR! stack     at C:\Users\Walter\Desktop\keira3\node_modules\sqlite3\node_modules\node-gyp\lib\find-visualstudio.js:71:14
npm error gyp ERR! stack     at C:\Users\Walter\Desktop\keira3\node_modules\sqlite3\node_modules\node-gyp\lib\find-visualstudio.js:384:16
npm error gyp ERR! stack     at C:\Users\Walter\Desktop\keira3\node_modules\sqlite3\node_modules\node-gyp\lib\util.js:54:7
npm error gyp ERR! stack     at C:\Users\Walter\Desktop\keira3\node_modules\sqlite3\node_modules\node-gyp\lib\util.js:33:16
npm error gyp ERR! stack     at ChildProcess.exithandler (node:child_process:422:5)
npm error gyp ERR! stack     at ChildProcess.emit (node:events:507:28)
npm error gyp ERR! stack     at maybeClose (node:internal/child_process:1101:16)
npm error gyp ERR! System Windows_NT 10.0.19045
npm error gyp ERR! command "D:\\Program Files\\nodejs\\node.exe" "C:\\Users\\Walter\\Desktop\\keira3\\node_modules\\sqlite3\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
npm error gyp ERR! cwd C:\Users\Walter\Desktop\keira3\node_modules\sqlite3
npm error gyp ERR! node -v v24.1.0
npm error gyp ERR! node-gyp -v v8.4.1
npm error gyp ERR! not ok
npm error A complete log of this run can be found in: C:\tmp\nodejs\npm-cache\_logs\2025-06-06T04_12_07_691Z-debug-0.log
```
</details>

https://github.com/azerothcore/Keira3/issues/3061

Helias commented on libs/features/creature/src/waypoint-data/waypoint-data.component.html:
```diff
@@ -0,0 +1,98 @@
+        <form [formGroup]="editorService.form" class="form-group edit-form">
+          <div class="row">
+            <div class="form-group col-12 col-sm-6 col-md-4 col-lg-3 col-xl-2">
+              <label class="control-label" for="point">Point</label>
```
not sure if we want to put here some LABEL for translations (like `LOADING` above)

pangolp replied:
Yes, no problem. The important thing is that everything else goes along and that they are the right files. I will see how to review the translations.

pangolp replied:
I have no problem adding spans for translations, but I wouldn't actually translate the attributes. What I would translate, in any case, is some kind of helper, such as a placeholder, to tell the person what that field indicates. But for now, you can leave it there, just like the rest of the tables.

Helias commented:
Did you run `npm install --global-production windows-build-tools` and did you restart the PC afterward?

pangolp commented:
> Did you run `npm install --global-production windows-build-tools` and did you restart the PC afterward?

I think not, I will review it.
Thank you.

pangolp commented:
Regarding the error, I'm updating Visual Studio, and if I remember correctly, when performing the installation, it recommends tools to work in node.js, which I probably won't install, because at the moment, I haven't had to use them, so I'm going to try adding those tools and using the command you sent me above.

pangolp commented:
When I run this command it seems to work, but I had to delete a file: `package-lock.json`
That file, however, is generated when the command is run.
But Git has it in the history, so it asks me to upload it.

```
$ git status
On branch waypoints_data
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   package-lock.json

no changes added to commit (use "git add" and/or "git commit -a")
```

<details>
<summary>npm install</summary>

```
C:\Users\Walter\Desktop\keira3>npm install
npm warn skipping integrity check for git dependency ssh://git@github.com/electron/node-gyp.git
npm warn deprecated @types/electron-settings@4.0.2: This is a stub types definition. electron-settings provides its own type definitions, so you do not need this installed.
npm warn deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm warn deprecated @npmcli/move-file@1.1.2: This functionality has been moved to @npmcli/fs
npm warn deprecated @npmcli/move-file@2.0.1: This functionality has been moved to @npmcli/fs
npm warn deprecated npmlog@6.0.2: This package is no longer supported.
npm warn deprecated @humanwhocodes/config-array@0.13.0: Use @eslint/config-array instead
npm warn deprecated rimraf@2.6.3: Rimraf versions prior to v4 are no longer supported
npm warn deprecated rimraf@2.7.1: Rimraf versions prior to v4 are no longer supported
npm warn deprecated har-validator@5.1.5: this library is no longer supported
npm warn deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm warn deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@8.1.0: Glob versions prior to v9 are no longer supported
npm warn deprecated glob@8.1.0: Glob versions prior to v9 are no longer supported
npm warn deprecated are-we-there-yet@3.0.1: This package is no longer supported.
npm warn deprecated @humanwhocodes/object-schema@2.0.3: Use @eslint/object-schema instead
npm warn deprecated gauge@4.0.4: This package is no longer supported.
npm warn deprecated boolean@3.2.0: Package no longer supported. Contact Support at https://www.npmjs.com/support for more info.
npm warn deprecated squel@5.13.0: No longer maintained
npm warn deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm warn deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm warn deprecated eslint@8.57.1: This version is no longer supported. Please see https://eslint.org/version-support for other options.
npm warn cleanup Failed to remove some directories [
npm warn cleanup   [
npm warn cleanup     'C:\\Users\\Walter\\Desktop\\keira3\\node_modules\\dmg-license\\node_modules\\json-schema-traverse',
npm warn cleanup     [Error: EPERM: operation not permitted, rmdir 'C:\Users\Walter\Desktop\keira3\node_modules\dmg-license\node_modules\json-schema-traverse\spec\fixtures'] {
npm warn cleanup       errno: -4048,
npm warn cleanup       code: 'EPERM',
npm warn cleanup       syscall: 'rmdir',
npm warn cleanup       path: 'C:\\Users\\Walter\\Desktop\\keira3\\node_modules\\dmg-license\\node_modules\\json-schema-traverse\\spec\\fixtures'
npm warn cleanup     }
npm warn cleanup   ]
npm warn cleanup ]

> Keira3@3.8.0 postinstall
> electron-builder install-app-deps

  • electron-builder  version=26.0.12
  • loaded configuration  file=C:\Users\Walter\Desktop\keira3\electron-builder.json
  • executing @electron/rebuild  electronVersion=34.5.3 arch=x64 buildFromSource=false appDir=./
  • installing native dependencies  arch=x64
  • preparing       moduleName=sqlite3 arch=x64
  • finished        moduleName=sqlite3 arch=x64
  • completed installing native dependencies

> Keira3@3.8.0 prepare
> husky install

husky - install command is DEPRECATED

added 2069 packages, and audited 2070 packages in 5m

298 packages are looking for funding
  run `npm fund` for details

24 vulnerabilities (22 moderate, 1 high, 1 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues possible (including breaking changes), run:
  npm audit fix --force

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.
```

</details>

pangolp commented:
I managed to run the project, but I don't see the table.
I must be missing it somewhere.

pangolp commented:
I would stick with translating the attributes within the component, although really, I don't know if it's a good thing, for me, the attributes have to be respected and remain in English, it would be good to add information in the last placeholder in the different languages, so that the person knows what they have to complete in that field, but if they prefer to translate the attribute, I have no problem doing so.

pangolp commented:
I think it should be removed from creature, because it seems to me that in this case, they are not related, although I don't really remember, perhaps the ID is the ID of the creature table, that is, the guid. The truth is, I haven't used the table for a long time.

Helias commented:
Let's keep the field name as is in the db, so in English.

About where to put the editor, let's move to a separate section like "Others ..."

FrancescoBorzi commented on libs/features/creature/src/waypoint-data/waypoint-data.component.html:
```diff
@@ -0,0 +1,98 @@
+                  <i [hidden]="!editorService.isRowSelected(row)" class="fas fa-chevron-right"></i>
+                </ng-template>
+              </ngx-datatable-column>
+              <ngx-datatable-column name="Point" prop="point" [minWidth]="60" [maxWidth]="80"></ngx-datatable-column>
+              <ngx-datatable-column name="Position X" prop="position_x" [minWidth]="80" [maxWidth]="120"></ngx-datatable-column>
+              <ngx-datatable-column name="Position Y" prop="position_y" [minWidth]="80" [maxWidth]="120"></ngx-datatable-column>
+              <ngx-datatable-column name="Position Z" prop="position_z" [minWidth]="80" [maxWidth]="120"></ngx-datatable-column>
+              <ngx-datatable-column name="Orientation" prop="orientation" [minWidth]="80" [maxWidth]="120"></ngx-datatable-column>
+              <ngx-datatable-column name="Delay" prop="delay" [minWidth]="60" [maxWidth]="80"></ngx-datatable-column>
+              <ngx-datatable-column name="Move Type" prop="move_type" [minWidth]="80" [maxWidth]="100"></ngx-datatable-column>
+              <ngx-datatable-column name="Action" prop="action" [minWidth]="60" [maxWidth]="80"></ngx-datatable-column>
+              <ngx-datatable-column name="Action Chance" prop="action_chance" [minWidth]="100" [maxWidth]="120"></ngx-datatable-column>
+              <ngx-datatable-column name="WP GUID" prop="wpguid" [minWidth]="80" [maxWidth]="100"></ngx-datatable-column>
```
empty compn

```suggestion
              <ngx-datatable-column name="Point" prop="point" [minWidth]="60" [maxWidth]="80" />
              <ngx-datatable-column name="Position X" prop="position_x" [minWidth]="80" [maxWidth]="120" />
              <ngx-datatable-column name="Position Y" prop="position_y" [minWidth]="80" [maxWidth]="120" />
              <ngx-datatable-column name="Position Z" prop="position_z" [minWidth]="80" [maxWidth]="120" />
              <ngx-datatable-column name="Orientation" prop="orientation" [minWidth]="80" [maxWidth]="120" />
              <ngx-datatable-column name="Delay" prop="delay" [minWidth]="60" [maxWidth]="80" />
              <ngx-datatable-column name="Move Type" prop="move_type" [minWidth]="80" [maxWidth]="100" />
              <ngx-datatable-column name="Action" prop="action" [minWidth]="60" [maxWidth]="80" />
              <ngx-datatable-column name="Action Chance" prop="action_chance" [minWidth]="100" [maxWidth]="120" />
              <ngx-datatable-column name="WP GUID" prop="wpguid" [minWidth]="80" [maxWidth]="100" />
```

pangolp commented:
> Thanks for your contribution! Can you please add the unit tests?

I'll try it. To do so, I need to delete the package-lock.json file, otherwise it won't download all the dependencies and won't start the project.

pangolp commented:
I need help with the last change that I have to make. The ID of this table would have to be, the guid of the creature, multiplied by 10. The problem, is that I am accessing from Creature_Template, not from Creature. Since in the waypoint_data, the creature's guid is used, and to that value, it is multiplied by 10. I am really lost. I don't know very well that I have to modify.
